sql/parser/eval.go

```diff
@@ -259,6 +263,158 @@ func (d DString) String() string {
 	return StrVal(d).String()
 }
 
+// These constants have been taken from the time package. There are a number
+// of approximations here because we don't care about accuracy.
+const (
+	secondsPerDay          = 24 * 60 * 60
+	internalToUnix   int64 = -(1969*366 + 1969/4 - 1969/100 + 1969/400) * secondsPerDay
+	maxSeconds             = math.MaxInt64 + internalToUnix
+	minSeconds             = math.MinInt64
+	maxSecondsInYear       = 366 * secondsPerDay
+	approxMaxYear          = maxSeconds - maxSecondsInYear
```

Comment:
This seems awfully specific to the internals of `time.Time`, which the comments in the `time` package suggest could change. The range of `time.Time` is fairly extraordinary. Couldn't we just some more reasonable limits like:

```
var (
  maxTime = time.Date(100000000, time.January, 1, 0, 0, 0, 0, time.UTC)
  minTime = time.Date(-100000000, time.January, 1, 0, 0, 0, 0, time.UTC)
)
```

This allows for dates 100m years into the future and 100m years into the past. Good enough for  everyone except astrophysicists and paleontologists.